src/reinforcement_learning.jl

```diff
@@ -0,0 +1,787 @@
+  configuration set via `set_rl_config!` before training.
+- `agent_types`: Single agent type (e.g., `MyAgent`) or vector of agent types (e.g., 
+  `[Predator, Prey]`) to train. All specified types must exist in the model and be 
+  listed in the RL configuration's `training_agent_types`.
+
+## Keyword Arguments  
+
+### Core Training Parameters
+
+- **`training_mode::Symbol`**: Training strategy (default: `:sequential`)
+  - `:sequential` - Train agent types one after another
+  - `:simultaneous` - Train all agent types together with alternating updates
+
+#### Sequential Training Steps
+This applies only when `training_mode=:sequential`:
+
+- **`training_steps::Int`**: Number of environment steps for training each agent type 
+  (default: 50,000). In sequential mode, this is per agent type. In simultaneous mode, 
+  this determines the batch size for each alternating update.
+
+#### Simultaneous Training Steps
+These apply only when `training_mode=:simultaneous`:
+
+- **`n_iterations::Int`**: Number of alternating training rounds (default: 5)
+- **`batch_size::Int`**: Size of training batches for each iteration (default: 10,000)
+
+## Algorithm Configuration
+
+- **`solver_params::Dict`**: Algorithm-specific hyperparameters. Can be:
+  - **Global parameters**: Applied to all agent types
+    ```julia
+    solver_params = Dict(
+        :ΔN => 200,              
+        :log => (period=1000,),
+    )
+    ```
+  - **Per-agent-type parameters**: Different settings for each agent type
+    ```julia
+    solver_params = Dict(
+        Predator => Dict(:ΔN => 100),
+        Prey => Dict(:ΔN => 200)
+    )
+    ```
+
+- **`solver_types::Dict{Type, Symbol}`**: Different RL algorithms for different agent types.
+  ```julia
+  solver_types = Dict(
+      FastAgent => :DQN,    
+      SmartAgent => :PPO      
+  )
+  ```
+
+## Network Architecture Customization
+
+- **`custom_networks::Dict{Type, Dict{Symbol, Function}}`**: Custom neural network 
+  architectures for specific agent types. Each entry maps an agent type to a dictionary 
+  containing `:value_network` and/or `:policy_network` functions.
+  ```julia
+  custom_networks = Dict(
+      MyAgent => Dict(
+          :value_network => () -> create_value_network((84,), [128, 64]),
+          :policy_network => () -> create_policy_network((84,), 5, action_space, [128, 64])
+      )
+  )
+  ```
+
+- **`custom_solvers::Dict{Type, Any}`**: Pre-configured complete solvers for specific 
+  agent types. Bypasses automatic solver creation.
+  ```julia
+  custom_solvers = Dict(
+      MyAgent => my_preconfigured_ppo_solver
+  )
+  ```
+
+
+## Returns
+- `ReinforcementLearningABM`: The input model with trained policies stored in 
+  `model.trained_policies`. The trained policies can be accessed via `get_trained_policies(model)` 
+  or copied to other models using `copy_trained_policies!(target, source)`.
+
+## Notes
+- `max_steps` is read directly from the RL configuration (`model.rl_config[][:max_steps]`)
```

Comment:
Why? This is odd. Why is this special training option as part of `rl_config` and all the others are given as keywords to the training function? We need to make up our mind instead of fragment these kind of decisions. I vote let's move this training option as yet another keyword to this function and remove it from `rl_config` alltogether.